server/aesdsocket.c

```diff
@@ -0,0 +1,201 @@
+/*
+* Author: Mubeena Udyavar Kazi
+* Course: ECEN 5713 - AESD
+* Reference: 
+*   1. ChatGPT with the following prompts.
+*      - Example for socket based server to send and receive data in C
+*      - Example in C to daemonize a process with signal handling
+*   2. Stack overflow
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <syslog.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+int sockfd, client_sockfd, datafd;
+
+void cleanup(int exit_code) {
+
+    // Close open sockets
+    if (sockfd >= 0) close(sockfd);
+    if (client_sockfd >=0) close(client_sockfd);
+
+    // Close file descriptors
+    if (datafd >= 0) close(datafd);
+
+    // Delete the file
+    remove("/var/tmp/aesdsocketdata");
+
+    // Close syslog
+    closelog();
+
+    // Exit
+    exit(exit_code);
+}
+
+void handle_signal(int sig) {
+   if (sig == SIGINT || sig == SIGTERM) {
+       syslog(LOG_INFO, "Caught signal, exiting");
+       cleanup(EXIT_SUCCESS);
```

Comment:
@mubeena12 are these functions safe to do from signal handler?  Consider a design where you can handle cleanup outside your signal handler instead.